File: sdk/go/internal/apiclient/v1/paths.go
// Package v1 holds wire-version-1 constants for the Go SDK.
//
// PathPrefix is the URL prefix every v1 route is published under on the
// sandbox daemon. When v2 lands, a sibling sdk/go/internal/apiclient/v2
// package will export its own PathPrefix, and the public Client config's
// APIVersion field selects which one to use.
//
// Keep this in sync with pkg/api/v1/dto.go::PathPrefix on the server.
package v1

const PathPrefix = "/v1"

File: sdk/typescript/src/internal/api/v1/paths.ts
// v1 path-prefix constant. The TypeScript SDK uses this to build URLs against
// the v1 surface of the sandbox daemon. When a new wire version lands, a
// sibling internal/api/vN/paths.ts will export its own PATH_PREFIX, and the
// client constructor's apiVersion option selects which one to use.
//
// Keep this in sync with pkg/api/v1/dto.go::PathPrefix on the server.
export const PATH_PREFIX = "/v1";

/**
 * Collection URL for a sandbox's custom-domain bindings. Used for both POST
 * (add) and GET (list). The DELETE variant lives at
 * {@link sandboxCustomDomainPath}.
 *
 * `id` is interpolated verbatim — sandbox IDs are server-generated opaque
 * tokens and already URL-safe.
 */
export function sandboxCustomDomainsPath(prefix: string, id: string): string {
  return `${prefix}/sandboxes/${id}/custom-domains`;
}

/**
 * URL for a single custom-domain binding. The hostname is percent-encoded so
 * IDN punycode (`xn--...`) and wildcard labels round-trip safely through
 * `DELETE /v1/sandboxes/{id}/custom-domains/{hostname}`.
 */
export function sandboxCustomDomainPath(prefix: string, id: string, hostname: string): string {
  return `${prefix}/sandboxes/${id}/custom-domains/${encodeURIComponent(hostname)}`;
}
